packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -742,6 +742,8 @@ mixin WidgetInspectorService {
   int _nextId = 0;
 
   List<String>? _pubRootDirectories;
+  /// Memoization for [_isLocalCreationLocation].
+  final Map<String, bool> _isLocalCreationCache = <String, bool>{};
```

Comment:
Faster would be an identity hash from 
Map<Uri, bool>
That will be significantly faster and as these objects are all const, you can trust identity hashes to be safe to use.

Reply:
This would require changing the kernel transform to write Uri's into the _Location fields.  We wouldn't want to parse the Uri in order to look up the cached result.  That would potentially wipe out our wins from the identityHash.  Since this is a complicated change, the kernel transform doesn't have any tests, and the performance written this way isn't bad, I'll leave that as a future optimization.

https://github.com/dart-lang/sdk/issues/48694

Reply:
A couple notes:
there are multiple tests in widget_inspector_test.dart that E2E test the kernel transformer. Adding some more tests there would be good. I'll add one to ensure the _Location fields are still const. If they are not, that is a serious bug.
Actually I was confused. The URI class isn't const so that wouldn't work as the _Location objects need to be const.